gui/build.py

```diff
@@ -79,25 +81,20 @@ def do_bundle(options):
     for dirname in ('build', 'dist'):
         if os.path.isdir(dirname):
             shutil.rmtree(dirname)
-    # create a intaller
-    python_lib_path = os.path.join(options.python_path, "Lib")
-    makensis_path = os.path.join(options.nsis_path, "makensis.exe")
+    # create a installer
     gui_path = os.path.realpath('.')
     moz_path = os.path.realpath('..')
-    python_dir = os.path.join(
-        os.path.dirname(sys.executable),
-        "Scripts"
-    )
-    args = []
-    args.append('--icon=wininst/app_icon.ico')
-    args.append('--base-name=Win32GUI')
-    args.append('--include-path=%s;%s;%s\site-packages;%s'
-                % (gui_path, moz_path, python_lib_path, python_lib_path))
-    args.append('--target-name=mozregression-gui.exe')
-    args.append('--target-dir=dist')
-    args.append('mozregui/main.py')
-    call(sys.executable, os.path.join(python_dir, 'cxfreeze'), *args)
-    call(makensis_path, 'wininst.nsi', cwd='wininst')
+    if IS_WIN:
+        python_lib_path = os.path.join(options.python_path, "Lib")
+        makensis_path = os.path.join(options.nsis_path, "makensis.exe")
+        call_cx_freeze(gui_path, moz_path, python_lib_path)
+        call(makensis_path, 'wininst.nsi', cwd='wininst')
+    else:
+        python_lib_path = options.python_path
+        call_cx_freeze(gui_path, moz_path, python_lib_path)
+        release_tar = tarfile.open('mozregression-gui.tar.gz', 'w:gz')
```

Comment:
A good pythonic alternative is to use the "with statement" (http://effbot.org/zone/python-with-statement.htm). It is safer because the tar file will be closed even if there is an issue - and it is shorter to write. Something like:
with tarfile.open('mozregression-gui.tar.gz', 'w:gz') as release_tar:
    release_tar.add(r'dist/')

You can look at the example section and search for the one using with: https://docs.python.org/2/library/tarfile.html#examples